File: src/main/java/com/neutralplasma/virtusbot/settings/NewSettings.java
package com.neutralplasma.virtusbot.settings;

import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.VoiceChannel;

import java.util.HashMap;

public class NewSettings {

    HashMap<String, String> stringData = new HashMap<>();
    HashMap<String, Long> longData = new HashMap<>();
    HashMap<String, Integer> intData = new HashMap<>();

    public NewSettings(){

    }

    public String getString(String setting){
        return stringData.get(setting);
    }

    public Long getLong(String setting){
        return longData.get(setting);
    }

    public int getInt(String setting){
        return intData.get(setting);
    }

    public void addIntData(String setting, int data){
        intData.put(setting, data);
    }

    public void addLongData(String setting, long data){
        longData.put(setting, data);
    }

    public void addStringData(String setting, String data){
        stringData.put(setting, data);
    }

}

File: src/main/java/com/neutralplasma/virtusbot/settings/NewSettings.kt
package com.neutralplasma.virtusbot.settings

data class NewSettings(
    var stringData: MutableMap<String, String>,
    var longData: MutableMap<String, Long>,
    var intData: MutableMap<String, Int> ){

    fun addStringData(path: String, data: String){
        this.stringData[path] = data;
    }
    fun addLongData(path: String, data: Long){
        this.longData[path] = data;
    }
    fun addIntData(path: String, data: Int){
        this.intData[path] = data;
    }

    fun getStringData(path: String): String{
        return this.stringData.getOrDefault(path, "none");
    }
    fun getLongData(path: String): Long{
        return this.longData.getOrDefault(path, 0);
    }
    fun getIntData(path: String): Int{
        return this.intData.getOrDefault(path, 0);
    }
}